server/src/main/java/org/apache/druid/client/coordinator/CoordinatorClient.java

```diff
@@ -95,13 +97,17 @@ public Boolean isHandOffComplete(String dataSource, SegmentDescriptor descriptor
   {
     try {
       FullResponseHolder response = druidLeaderClient.go(
-          druidLeaderClient.makeRequest(HttpMethod.GET,
-                                        StringUtils.format(
-                                            "/druid/coordinator/v1/datasources/%s/intervals/%s/serverview?partial=%s",
-                                            dataSource,
-                                            interval.toString().replace('/', '_'),
-                                            incompleteOk
-                                        ))
+          druidLeaderClient.makeRequest(
+              HttpMethod.GET,
+              StringUtils.format(
+                  "/druid/coordinator/v1/datasources/%s/intervals/%s/serverview?partial=%s",
+                  // FIXME should this get url encoded while I'm at it? It was definitely needed
```

Comment:
What do you mean by encoding url here?

Reply:
Hmm, it's confusing that IntelliJ reformatted this part, which is unchanged other than my comment. I do think my settings are based on the Druid configuration...

In my new method below (getDatabaseSegmentDataSourceSegments) it was necessary for me to write `StringUtils.urlEncode(dataSource)` in order to get the integration tests (which helpfully put a bunch of non-ASCII characters in the data source name) to pass.

`isHandOffComplete` above also does that. But this method does not. So that makes me think that this method has a bug (unrelated to the work in this PR).

Reply:
Oh, yeah `dataSource` should be encoded. Thanks for catching it!